### PR TITLE
Blankbot cmake modernization

### PR DESCRIPTION
**Pull request by ghost**
Depends on https://github.com/cpp-sc2/cpp-sc2/pull/118

Not for immediate merging due to dependency, and the `GIT_REPOSITORY` and `GIT_TAG` for `cpp-sc2` are currently pointing to my fork commit for the dependent PR. Once the `cpp-sc2` PR is approved and merged, I will rebase the correct `FetchContent` configuration.

Otherwise this PR is for review of changes made to bring `BlankBot` in line with the incoming `cpp-sc2` build configuration changes.

### CODE REVIEW

**Comment by ghost** [comment]
Waiting on https://github.com/cpp-sc2/cpp-sc2/pull/121 to be merged to set the `cpp-sc2` `FetchContent` commit. Once rebased the PR will be ready for merge.

File: cmake/cpp_sc2.cmake
Comment (alkurbatov) [inline cmake/cpp_sc2.cmake]: Please use the same folders structure as in cpp-sc2, i.e. this one should be thirdparty/... as cpp-sc2 is external dependency for this bot.

File: src/CMakeLists.txt
Comment (alkurbatov) [inline src/CMakeLists.txt]: Should be `endif ()`.

File: src/CMakeLists.txt
Comment (alkurbatov) [inline src/CMakeLists.txt]: Typo: `UNIT` -> `UNIX`.

Also:
* If we use `elseif` there is no need to doublecheck `NOT APPLE`.
* Please keep the original comment regarding Linux, it is kept mostly for people less experienced with cmake.

File: CMakeLists.txt
Comment (alkurbatov) [inline CMakeLists.txt:33]: Style: Please, finish the comments with dots for consistency (or drop dots everywhere).
Here and in the other places.

File: README.md
Comment (alkurbatov) [inline README.md]: In this case, examples should use `-B build` as we are outside of the build folder.
Here and below.

File: README.md
Comment (alkurbatov) [inline README.md]: I believe this list of options is unnecessary because we already have detailed description below.
I would prefer to keep it in one place to reduce the maintenance cost.

**Review by alkurbatov — CHANGES_REQUESTED** [review]
Looks good but needs minor corrections.

**Comment by alkurbatov** [comment]
Btw, it seems that the PR has passed on ubuntu-18.04. Did they decided to extend support of the distro?

**Comment by ghost** [comment]
Yes as per [the original announcement](https://github.com/actions/runner-images/issues/6002), actions will fail during brownouts (like on the original deprecation start date) and the image will be fully deprecated in December.

File: README.md
Comment (alkurbatov) [inline README.md]: Missing `-B build`.

**Comment by alkurbatov** [comment]
Looks very good, only one fix  left (in README).

File: README.md
Comment (alkurbatov) [inline README.md]: Looks like the `build` part is still missing.

**Comment by ghost** [comment]
Maybe hold off on merge to rebase `cpp-sc2` commit hash to https://github.com/cpp-sc2/cpp-sc2/pull/122 once merged, as WSL2 options will fail here as well on CMake versions older than 3.24.

**Comment by alkurbatov** [comment]
Thank you! Very good work.

**Pull request merged**
